test/unit/app/controllers/permissions/permissions-middleware-test.js

```diff
@@ -0,0 +1,752 @@
+      )
+
+      const id = permController.pendingApprovals.keys().next().value
+
+      await permController.rejectPermissionsRequest(id)
+      await requestRejection
+
+      assert.ok(
+        (
+          !res.result && res.error &&
+        res.error.message === expectedError.message
+        ),
+        'response should have expected error and no result'
+      )
+
+      const aAccounts = await permController.getAccounts(ORIGINS.a)
+      assert.deepEqual(
+        aAccounts, [], 'origin should have have correct accounts'
+      )
+    })
+
+    it('directly returns accounts for permitted domain', async function () {
+
+      const cMiddleware = getPermissionsMiddleware(permController, ORIGINS.c)
+
+      grantPermissions(
+        permController, ORIGINS.c,
+        PERMS.finalizedRequests.eth_accounts(ACCOUNT_ARRAYS.c)
+      )
+
+      const req = RPC_REQUESTS.eth_requestAccounts(ORIGINS.c)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(
+        res.result && !res.error,
+        'response should have result and no error'
+      )
+      assert.deepEqual(
+        res.result, ACCOUNT_ARRAYS.c,
+        'response should have correct result'
+      )
+    })
+  })
+
+  describe('wallet_sendDomainMetadata', function () {
+
+    let permController
+
+    beforeEach(function () {
+      permController = initPermController()
+    })
+
+    it('records domain metadata', async function () {
+
+      const name = 'BAZ'
+
+      const cMiddleware = getPermissionsMiddleware(permController, ORIGINS.c)
+
+      const req = RPC_REQUESTS.wallet_sendDomainMetadata(ORIGINS.c, name)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(res.result, 'result should be true')
+
+      const metadataStore = permController.store.getState()[METADATA_STORE_KEY]
+
+      assert.deepEqual(
+        metadataStore,
+        { [ORIGINS.c]: { name, extensionId: undefined } },
+        'metadata should have been added to store'
+      )
+    })
+
+    it('records domain metadata and preserves extensionId', async function () {
+
+      const extensionId = 'fooExtension'
+
+      const name = 'BAZ'
+
+      const cMiddleware = getPermissionsMiddleware(permController, ORIGINS.c, extensionId)
+
+      const req = RPC_REQUESTS.wallet_sendDomainMetadata(ORIGINS.c, name)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(res.result, 'result should be true')
+
+      const metadataStore = permController.store.getState()[METADATA_STORE_KEY]
+
+      assert.deepEqual(
+        metadataStore,
+        { [ORIGINS.c]: { name, extensionId } },
+        'metadata should have been added to store'
+      )
+    })
+
+    it('should have record domain metadata if no name', async function () {
```

Comment:
Is this what was meant? I'm not entirely sure what this test is doing if not.
```suggestion
    it('should not record domain metadata if no name', async function () {
```

Reply:
Fixed in 138fc54